examples/tvos/objc/DownloadCache/RepositoriesViewController.m

```diff
@@ -0,0 +1,119 @@
+//
+//  RepositoriesViewController.m
+//  DownloadCache
+//
+//  Created by Katsumi Kishikawa on 11/19/15.
+//  Copyright © 2015 Realm. All rights reserved.
+//
+
+#import "RepositoriesViewController.h"
+#import "RepositoryCell.h"
+#import "Repository.h"
+
+@interface RepositoriesViewController () <UITextFieldDelegate>
+
+@property (nonatomic, weak) IBOutlet UISegmentedControl *sortOrderControl;
+@property (nonatomic, weak) IBOutlet UITextField *searchField;
+
+@property (nonatomic) RLMResults *results;
+
+@end
+
+@implementation ViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+
+    NSURLComponents *components = [NSURLComponents componentsWithString:@"https://api.github.com/search/repositories"];
+    components.queryItems = @[[NSURLQueryItem queryItemWithName:@"q" value:@"language:objc"],
+                              [NSURLQueryItem queryItemWithName:@"sort" value:@"stars"],
+                              [NSURLQueryItem queryItemWithName:@"order" value:@"desc"]];
+    [[[NSURLSession sharedSession] dataTaskWithURL:components.URL completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+        if (!error) {
+            NSError *jsonError = nil;
+            NSDictionary *repositories = [NSJSONSerialization JSONObjectWithData:data options:kNilOptions error:&jsonError];
+            if (!jsonError) {
+                NSArray *items = repositories[@"items"];
+
+                RLMRealm *realm = [RLMRealm defaultRealm];
+                [realm transactionWithBlock:^{
+                    for (NSDictionary *item in items) {
+                        Repository *repository = [Repository new];
+                        repository.identifier = [NSString stringWithFormat:@"%@", item[@"id"]];
+                        repository.name = item[@"name"];
+                        repository.avatarURL = item[@"owner"][@"avatar_url"];
+
+                        [realm addOrUpdateObject:repository];
+                    }
+                }];
+                dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
This hits the race condition where the async dispatch arrives before the refresh. Reloading in a realm notification block instead would fix that and would also reduce nesting.